DFE-Digital/schools-experience
# Se 1558/schools/on boarding/ab test candidate requirements

### Context

The requirement is to show some % of schools the new
candidate_requirements_selection screen comprised of checkboxes and the
remaining % of schools the old candidate_requirement free text area.
To determine which requirements the school enters we set a flag on the
school profile when it is created and use this flag to determine how the
school progresses through the wizard.

### Changes proposed in this pull request
Adds the new candidate requirements screen.
Adds feature flag to enable the new journey.

### Guidance to review
It says 45 files changed but 15 of them are just updating the feature specs to send candidates on the new journey!
Check the logic in current_step is sensible and there's no way for a school to see a different journey than the one they first completed the wizard with.

Manual testing if required:
Set AB_TEST_THRESHOLD environment variable to 100 complete the school on boarding wizard, you should see the new journey.
Set AB_TEST_THRESHOLD to 0 complete the school on boarding wizard expecting to see the old candidate requirement screen.
